packages/admin/src/pages/build-application/[applicationId]/[sectionId]/index.test.tsx

```diff
@@ -0,0 +1,205 @@
+        redirect: {
+          destination:
+            '/service-error?serviceErrorProps={"errorInformation":"Something went wrong while trying to edit a section","linkAttributes":{"href":"/scheme-list","linkText":"Please find your scheme application form and continue.","linkInformation":"Your previous progress has been saved."}}',
+          permanent: false,
+        },
+      });
+    });
+
+    it('Should redirect to service error page if something goes wrong', async () => {
+      mockedGetApplicationFormSummary.mockRejectedValueOnce(new Error('error'));
+
+      const result = await getServerSideProps(getContext(getDefaultContext));
+
+      expectObjectEquals(result, {
+        redirect: {
+          destination:
+            '/service-error?serviceErrorProps={"errorInformation":"Something went wrong while trying to edit a section","linkAttributes":{"href":"/scheme-list","linkText":"Please find your scheme application form and continue.","linkInformation":"Your previous progress has been saved."}}',
+          permanent: false,
+        },
+      });
+    });
+  });
+
+  describe('Edit Section Page', () => {
+    beforeEach(() => {
+      render(<EditSectionPage {...getPageProps(getDefaultProps)} />);
+    });
+
+    it('Should render a meta title', () => {
+      expect(document.title).toBe('Build an application form - Manage a grant');
+    });
+
+    it("Should render 'Back' button", () => {
+      expect(screen.getByRole('link', { name: 'Back' })).toHaveAttribute(
+        'href',
+        '/apply/build-application/some-application-id/dashboard'
+      );
+    });
+
+    it('Should render title of the page', () => {
+      screen.getByText('Edit this section');
+    });
+
+    it('Should render the section title', () => {
+      screen.getByText('some-section-title');
+    });
+
+    it('Should render a question', () => {
+      screen.getByText('some-question-title');
+      screen.getByText('Long answer');
+      expect(screen.getByRole('link', { name: 'Edit' })).toHaveProperty(
+        'href',
+        'http://localhost/apply/#'
+      );
+    });
+
+    it('Should render a button to add a new question', () => {
+      expect(
+        screen.getByRole('button', { name: 'Add a new question' })
+      ).toHaveAttribute(
+        'href',
+        '/apply/build-application/some-application-id/testSectionId/question-content'
+      );
+    });
+
+    it('Should render a button to delete this section', () => {
+      expect(
+        screen.getByRole('button', { name: 'Delete section' })
+      ).toHaveAttribute(
+        'href',
+        '/apply/build-application/some-application-id/testSectionId/delete-confirmation'
+      );
+    });
+
+    it('Should render a button to save this section', () => {
+      expect(
+        screen.getByRole('button', { name: 'Save and go back' })
+      ).toHaveAttribute(
+        'href',
+        '/apply/build-application/some-application-id/dashboard'
+      );
+    });
```

Comment:
```suggestion
it.each([ 
  ["Add a new question", “/apply/build-application/some-application-id/testSectionId/question-content”], 
  ["Delete section", “/apply/build-application/some-application-id/testSectionId/delete-confirmation”],
  ["Save and go back", “/apply/build-application/some-application-id/dashboard”]
])(“button %s should have href %s”, 
 (name, href) => { 
    expect(screen.getByRole('button', { name })).toHaveAttribute('href', href);
  }
);
```

Reply:
Neat - didnt know you could do this <3 formatter kinda rips it to shreds, but still worth it

Reply:
yeah test.each is great for this kind of thing and doesn't seem widely known
https://elfi-y.medium.com/reduce-boilerplate-test-code-with-jest-it-each-30a0eec9776d